src/main/kotlin/org/wycliffeassociates/otter/common/device/IAudioPlayer.kt

```diff
@@ -7,7 +7,7 @@ interface IAudioPlayer {
 
     fun load(file: File): Completable
 
-    fun play()
+    fun play(onFinish: (() -> Unit) = {})
```

Comment:
Weird way to register an onCompleteListener. Problematic if multiple things wish to register one. Also, should it only be called once, of every time that play is finished?

Reply:
The audio player presented an interesting case since we want to use a singleton. We have multiple things that could use the same audio player, and they all only need to listen when `play` is called (they could start at `load`, but no listener events would happen between load and play). If they register to listen earlier, they could end up listening to events generated when someone else is using the audio player (I discovered that quirk when testing what happens when you show two audio cards at the same time). So the player should really only be "valid" for a particular listener if that listener was the one that initiated the playback.

I ended up with this solution, but we could switch it up to have an addCompleteListener function in the player. However, to make sure everything behaves as expected, we would also need to clear all the listeners each time `load` is called. Otherwise we have audio cards listening to the audio events from other audio cards.